docs/client_retries.md

```diff
@@ -0,0 +1,321 @@
+# Client Side Retries
+Client Libraries use retries to handle unexpected, transient failures (i.e. server is temporarily unavailable).
+Multiple attempts, hopefully, will result in a successful response from the server.
+
+Default retry values are selected by the team operating the cloud service. These retry values are configured
+per RPC. A service *may* choose to only enable retries for a subset of RPCs. It is possible that each RPC for
+a service is configured differently.
+
+## Retry Parameters
+Client libraries have two types of retry parameters to configure:
+1. Retry Status Code: Set of status codes to retry on
+2. Retry Timeout/ Attempt Bounds: Configurable [RetrySettings](https://cloud.google.com/java/docs/reference/gax/latest/com.google.api.gax.retrying.RetrySettings) to define the bounds
+
+### Default RPC Retry Configuration Location
+The default retry configurations are defined in the generated `{Client}StubSettings` file. Using the ExportAssets RPC in
+Java-Asset v3.41.0 as an example, the default retry configurations are defined in the following places:
+<br>
+- Retry Status Codes are configured [here](https://github.com/googleapis/google-cloud-java/blob/d9da511b4b56302e509abe8b2d919a15ea7dcae7/java-asset/google-cloud-asset/src/main/java/com/google/cloud/asset/v1/stub/AssetServiceStubSettings.java#L1058-L1082). Example:
+```java
+ImmutableMap.Builder<String, ImmutableSet<StatusCode.Code>> definitions = ImmutableMap.builder();
+definitions.put("no_retry_0_codes", ImmutableSet.copyOf(Lists.<StatusCode.Code>newArrayList()));
+// ... More StatusCode configurations
+RETRYABLE_CODE_DEFINITIONS = definitions.build();
+```
```

Comment:
nit: consider indenting these snippets by ~4 spaces to make them render as indented under their parent bullet.

Reply:
Done